test/__snapshots__/test/test.ts.md

```diff
@@ -1861,6 +1861,92 @@ Generated by [AVA](https://avajs.dev).
     }␊
     `
 
+## intersection.4.js
+
+> Expected output to match snapshot for e2e test: intersection.4.js
+
+    `/* eslint-disable */␊
+    /**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    export type Intersection = A | B;␊
+    export type A = A1 & {␊
+      b?: B;␊
+    };␊
+    export type A1 = Base;␊
+    export type B = B1 & {␊
+      x?: string;␊
+    };␊
+    export type B1 = Base;␊
+    ␊
+    export interface Base {␊
+      y?: string;␊
+    }␊
+    `
+
+## intersection.5.js
+
+> Expected output to match snapshot for e2e test: intersection.5.js
+
+    `/* eslint-disable */␊
+    /**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    export type Intersection = A | B;␊
+    export type A = Base & {␊
+      b?: B;␊
+      [k: string]: unknown;␊
+    };␊
+    export type B = Base & {␊
+      x?: string;␊
+      [k: string]: unknown;␊
+    };␊
+    ␊
+    export interface Base {␊
+      y?: string;␊
+    }␊
+    `
+
+## intersection.6.js
+
+> Expected output to match snapshot for e2e test: intersection.6.js
+
+    `/* eslint-disable */␊
+    /**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    export type Intersection = Car | Truck;␊
+    export type Car = Car1 & {␊
```

Comment:
And here. This output is cleaner, if you can achieve something similar with your approach: https://github.com/bcherny/json-schema-to-typescript/pull/605/files#diff-34615f003eb49f90c3044b8fafebe3fe6b162e290b9b724082a6d72f58cfa4afR1927